scripts/prepare-release.mjs

```diff
@@ -0,0 +1,350 @@
+		);
+	}
+	return {
+		originalBranchName: currentBranchName,
+		releaseBranch: branchName,
+	};
+}
+
+/**
+ * Set new version in the main plugin file.
+ * This method also creates a commit in the current branch.
+ *
+ * @param {string} filename     The path to the main plugin file.
+ */
+function updateVersionInFile( filename ) {
+	console.log( 'Updating plugin file versions ...' );
+	let newPluginFileContents = readFileContents( filename ).replace(
+		/(Version|Stable tag): (.*)/,
+		`$1: ${ version }`,
+	);
+
+	// Update version constant.
+	const { constant }    = PLUGINS[ pluginSlug ];
+	newPluginFileContents = newPluginFileContents.replace(
+		new RegExp( `define\( '${constant}', '.*' \);` ),
+		`define( '${ constant }', '${ version }' );`,
+	);
+
+	fs.writeFileSync( filename, newPluginFileContents, 'utf-8' );
+	execSync(
+		`git add ${ filename }`,
+	);
+}
+
+/**
+ * Replaces the next-version placeholder with the new version to be released.
+ * This method also creates a commit in the current branch.
+ */
+function replaceNextVersionPlaceholder() {
+	console.log( `Replacing next version placeholder with ${ version } ...` );
+	execSync( `bash scripts/replace-next-version-tag.sh ${ version }` );
+	execSync(
+		`git add .`,
+	);
+}
+
+/**
+ * Update package.json and package-lock.json files with the new version to be released.
+ * This method also creates a commit in the current branch.
+ */
+function updatePackageJsonFiles() {
+	console.log( 'Updating package.json version...' );
+	try {
+		execSync( `npm version ${ version } --no-git-tag-version` );
+	} catch {
+		throw new Error( 'Version could not be updated in package.json file.' );
+	}
+
+	execSync(
+		`git add package.json package-lock.json`,
+	);
+}
+
+/**
+ * Generate POT files (translations).
+ * This method also creates a commit in the current branch.
+ */
+function generatePotFiles() {
+	console.log( 'Updating POT files...' );
+	try {
+		execSync( `npm run i18n:build 2> /dev/null` );
+	} catch {
+		throw new Error( 'POT file generation failed.' );
+	}
+	execSync( `git add languages/` );
+}
+
+function commitFiles() {
+	execSync( `git commit -m "Update plugin to ${ version }."` );
+}
+
+/**
+ * Generates the changelog based on the PRs.
+ * This method also creates a commit in the current branch.
+ */
+function buildReleaseNotes() {
+	let prs = execSync( `${ ghPrs }  --json number,title,body,labels` );
+	prs     = JSON.parse( prs );
+
+	const changelogs = prs.map( ( pr ) => {
+		const body              = pr.body;
+		const changelogSections = body.match( /### Release Notes([\S\s]*?)(?:###|<!--|$)/ );
+
+		if ( ! changelogSections ) {
+			return `* ${ pr.title } (#${ pr.number })`;
+		}
+		const changelog = changelogSections[ 1 ].trim();
+
+		if ( ! changelog.match( /\w/ ) ) {
+			return '';
+		}
+
+		return changelog;
+	} );
+
+	console.log( 'Proposed changelog: ' );
+	console.log( changelogs.join( "\n" ) );
+
+	return changelogs.join( "\n" );
+}
```

Comment:
Wouldn't it be better to just use [jetpack's changelogger](https://github.com/Automattic/jetpack-changelogger) instead?

It is already used on Sensei, Sensei Pro and so on.. :thinking:

Reply:
I wrote about this more on P2 (p6r3EZ-1Py-p2) — this is a bit of an experiment, the bet is that writing the changelog entries in the PR editor should be a better developer experience and produce better quality changelogs. I also want to expand this later to generating a developer notes section from the "Updated hooks/templates" section we have in PRs.